lib/internal/streams/from.js

```diff
@@ -37,21 +37,34 @@ function from(Readable, iterable, opts) {
   // Reading boolean to protect against _read
   // being called before last iteration completion.
   let reading = false;
+  let needToClose = false;
   readable._read = function() {
     if (!reading) {
       reading = true;
       next();
     }
   };
+  readable._destroy = function() {
+    if (needToClose) {
+      needToClose = false;
+      if (typeof iterator.return === 'function') {
+        iterator.return();
```

Comment:
You need to await and wrap this in a try/catch as well. As `return()` can fail.

Reply:
Hm. Ok. But then I also need to await `.value` returned from `.return`.
Though I'm not sure how to do all this in sync `_destroy` method.

Reply:
`_destroy` is not sync

Reply:
You are missing stuff here. The signature is `_destroy(err, cb)`.

Reply:
>  I also need to await .value returned from .return.

No, I don't think you need to. I don't think `return()` is supposed to return anything?

Reply:
```js
async _destroy(err, cb) {
  try {
    if (needToClose && typeof iterator.return === 'function') {
      await iterator.return();
    }
    cb();
  } catch (er) {
    cb(err || er);
  }
}